api/client/network/create.go

```diff
@@ -55,6 +56,7 @@ func newCreateCommand(dockerCli *client.DockerCli) *cobra.Command {
 	flags.StringSliceVar(&opts.labels, "label", []string{}, "Set metadata on a network")
 	flags.BoolVar(&opts.internal, "internal", false, "Restrict external access to the network")
 	flags.BoolVar(&opts.ipv6, "ipv6", false, "Enable IPv6 networking")
+	flags.BoolVar(&opts.manuallyAttachable, "manually-attachable", false, "Enable manual container attachment")
```

Comment:
@cpuguy83 The flag's existence is purely for security reasons.

Background: Networks are **completely** unreachable from nodes that are not running any service from that network (e.g. they don't gossip and they don't have the encryption keys). Isolation can be achieved for instance by limiting a service to a specific set of nodes through constraints.

However, there are (many) use cases that require manually attaching a container to a network using docker run. In order to keep our secure by default mantra, it was decided to have this as a separate flag.

We could think this through and perhaps switch the flag to on by default in future releases depending on user reaction.

/cc @diogomonica @icecrime